WordPress/openverse-api
# Fix broken load script and port to Python

## Fixes


Fixes #828 by @dhruvkb 

## Description


This PR 
1. fixes the issue in the `load_sample_data.sh` script that caused the loop to run 3 times without ever breaking (commit fde670f)
2. ports the script to Python (making the above point redundant), thereby adding many new features and capabilities

The Python script has the following advantages:
- **idempotency**: This allows you to reset everything by running `just init` as many times as you want. There is more scope to improve but it will never crash on re-runs.
- **clarity**: The output of Bash scripts can be messy and out of control (e.g. `set -x`). With Python logging levels, you can tweak how much of a mess you want in the output.
- **DX friendliness**: Python is leaps and bounds ahead of Bash in terms of ease of use (subjective). The Python script is well documented, modular and easy to update.
- **testable**: It might be overkill but you can write unit tests for the script's functions. 

Oh, and you can drop into a shell and import and run the functions individually.

## Side-effects

To have a Python file in the root of the repo, we need to have a Python workspace at the root of the repo. In a separate PR we can delete `pre-commit` and `pycodestyle` from the inner packages.

In the future, we can consider moving the documentation tooling (Sphinx & friends) out of the API into the root as well.

## Testing Instructions

The passing CI should be sufficient. It means the load script is doing it's job. As for idempotency, follow these steps:
0. Check out the PR.
1. Repeatedly run `just init`.
2. Observe that there are no errors.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
